src/bootstrap/bootstrap.py

```diff
@@ -431,6 +431,19 @@ def support_xz():
                 with output(self.rustfmt_stamp()) as rustfmt_stamp:
                     rustfmt_stamp.write(self.date + self.rustfmt_channel)
 
+        if self.downloading_llvm():
+            llvm_sha = subprocess.check_output(["git", "log", "--author=bors",
+                "--format=%H", "-n1"]).decode(sys.getdefaultencoding()).strip()
+            llvm_assertions = self.get_toml('assertions', 'llvm') == 'true'
+            if self.program_out_of_date(self.llvm_stamp(), llvm_sha + str(llvm_assertions)):
```

Comment:
To make sure I understand this right, this means that each time you rebase you'll re-download LLVM?

Reply:
Currently, yes. Once a submodule bump of llvm-project lands I plan to make this "rebase onto a new version of LLVM"; that has not happened yet. Since it's a relatively small download (~20MB) I'd personally hope to land this now with the less ideal implementation and then adjust it to look for the most recent submodule bump in your history.

Reply:
Ok seems reasonable, perhaps an issue can be filed about this too?

Reply:
Filed https://github.com/rust-lang/rust/issues/76587.